h5py/h5t.pyx

```diff
@@ -1025,23 +1056,7 @@ cdef class TypeFloatID(TypeAtomicID):
 
         # Handle non-standard exponent and mantissa sizes.
         for ftype_, finfo, size in _available_ftypes:
-            nmant = finfo.nmant
-            maxexp = finfo.maxexp
-            minexp = finfo.minexp
-            # workaround for numpy's buggy finfo on float128 on ppc64 archs
-            if ftype_ == np.longdouble and MACHINE == 'ppc64':
-                # values reported by hdf5
-                nmant = 116
-                maxexp = 1024
-                minexp = -1022
-            elif ftype_ == np.longdouble and MACHINE == 'ppc64le':
-                # values reported by hdf5
-                nmant = 52
-                maxexp = 1024
-                minexp = -1022
-            elif nmant == 63 and finfo.nexp == 15:
-                # This is an 80-bit float, correct mantissa size
-                nmant += 1
+            nmant, maxexp, minexp = _correct_float_info(ftype_, finfo)
```

Comment:
Are there any concerns about calling a Python function `def _correct_float_info` from inside a `cdef` function. I presume Cython will make it work, but could it affect performance or threading behaviour significantly?